sljit_src/sljitExecAllocator.c

```diff
@@ -192,6 +192,63 @@ static SLJIT_INLINE void free_chunk(void *chunk, sljit_uw size)
 
 #endif
 
+#if defined(__APPLE__) && defined(__arm64__)
+__thread jit_protect_mode arm_current_jit_protect_mode = JPM_ENABLED;
```

Comment:
this will likely break if not linking with pthread (should also `#define SLJIT_SINGLE_THREADED 1` in sljitConfig.h

Reply:
I hope this patch will work without remembering the previous state, so this problem should go away.